var/spack/repos/builtin/packages/metplus/package.py

```diff
@@ -33,10 +34,18 @@ class Metplus(Package):
     variant("makeplots", default=False, description="Enable MakePlots Wrapper.")
     variant("plotdataplane", default=False, description="Generate images from Postscript output.")
 
+    depends_on("python@3.10.4:", when="@6:", type=("run"))
+    depends_on("met@12:+python", when="@6:", type=("run"))
```

Comment:
(Stylistic, optional) More common ordering is to have the generic dependency first, then the version constrained dependencies, i.e. this would go below the next line. Applies to other additions too.